Service fails to auto-start at logon: plug-in DLLs depend on DisplayXRClient.dll which is not on the loader's DLL search path (err=126)

## Summary

`displayxr-service.exe` exits immediately (`-1`) at Windows logon, so DisplayXR never auto-starts after a reboot. Root cause: the display-processor plug-in DLLs statically import `DisplayXRClient.dll`, which lives **only** in `…\DisplayXR\Runtime\` and is **not on PATH**. The plug-in loader resolves that dependency only via the *current-working-directory* DLL search step. The `HKLM\…\Run\DisplayXR Service` entry launches the service with cwd = `C:\Windows\System32`, so the dependency is not found, **both** plug-ins fail `LoadLibrary` with **err=126 (ERROR_MOD_NOT_FOUND)**, no display processor is created, and init aborts.

This reproduces on a clean install from `DisplayXRBundle-0.3.0.exe` (runtime v1.5.0, leia-plugin v1.0.2).

## Symptom / service log

```
[WARN ] [try_load_one] plugin loader:   leia-sr: LoadLibrary(C:\Program Files\DisplayXR\Plugins\LeiaSR\DisplayXR-LeiaSR.dll) failed (err=126).
[WARN ] [try_load_one] plugin loader:   sim-display: LoadLibrary(C:\Program Files\DisplayXR\Runtime\plugins\DisplayXR-SimDisplay.dll) failed (err=126).
[WARN ] [discover_active_plugin] plugin loader: no registered plug-in claimed the system — falling back to static drivers.
[ERROR] [sim_display_open_system_impl] sim_display builder: no display-processor plug-in created a device.
[ERROR] [init_all] xrt_instance_create_system failed: XRT_ERROR_DEVICE_CREATION_FAILED
[ERROR] [ipc_server_main] init_all failed: XRT_ERROR_DEVICE_CREATION_FAILED
```

Note: even **`sim-display`** (the vendor-neutral in-tree plug-in, no SR dependency) fails — so this is not a Leia/SR-Platform PATH problem.

## Root cause

Both plug-ins import the runtime core DLL:

```
DisplayXR-SimDisplay.dll  ->  DisplayXRClient.dll, D3DCOMPILER_47.dll, d3d12.dll, …
DisplayXR-LeiaSR.dll      ->  DisplayXRClient.dll, SimulatedReality*.dll, d3d11/12.dll, …
```

`DisplayXRClient.dll` exists only at `C:\Program Files\DisplayXR\Runtime\DisplayXRClient.dll`. It is **not** copied next to either plug-in (`Plugins\LeiaSR\`, `Runtime\plugins\`) and `…\DisplayXR\Runtime` is **not** on the machine PATH. The plug-in loader appears to load plug-ins with a search mode that drops the application (exe) directory from the dependency search (e.g. `LOAD_WITH_ALTERED_SEARCH_PATH` / `LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR`), so `DisplayXRClient.dll` is found **only** when the process CWD happens to be `Runtime\`.

## Evidence — A/B/C by working directory (PATH identical in all three)

| Working dir | Plug-in load | Service |
|---|---|---|
| `…\DisplayXR\Runtime` | ✅ `active plug-in: id=leia-sr` | stays alive |
| `C:\Windows\System32` (logon Run-key default) | ❌ both `err=126` | **dies, exit -1** |
| arbitrary other dir | ❌ `err=126` | **dies, exit -1** |

`dumpbin /dependents` confirms the `DisplayXRClient.dll` import on both plug-ins; `DisplayXRClient.dll` is present only in `Runtime\`; Runtime dir is absent from PATH.

## Regression origin

Introduced by the vendor-plug-in extraction (ADR-019 / #256 / #263). Before that the display processor was in-process (no separate plug-in DLL, no cross-directory dependency), so logon auto-start worked. Same dependency-resolution class as the (closed) #314 Leia VK delay-load bug.

## Suggested fix (loader side)

Make the runtime install dir part of the plug-in loader's DLL search path before `LoadLibrary`, independent of CWD. Options:

- `SetDefaultDllDirectories(LOAD_LIBRARY_SEARCH_DEFAULT_DIRS)` + `AddDllDirectory(<runtime_exe_dir>)`, then load with `LOAD_LIBRARY_SEARCH_USER_DIRS`; **or**
- pre-load `DisplayXRClient.dll` by absolute path before enumerating plug-ins; **or**
- load each plug-in with `LOAD_LIBRARY_SEARCH_APPLICATION_DIR | LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR` (so both the exe dir *and* the plug-in dir are searched).

(A `SetDllDirectory(exe_dir)` shim would also work but is process-global and less precise.)

## Environment

- Windows 11 Pro 26200
- Fresh install via `DisplayXRBundle-0.3.0.exe` → Runtime **v1.5.0**, Leia plug-in **v1.0.2**, sim-display v1.5.0
- Reproduced by launching `displayxr-service.exe` directly with cwd ≠ `Runtime\`; matches user-reported "service did not auto-start after reboot."

--- a/src/xrt/targets/common/target_plugin_loader.c
+++ b/src/xrt/targets/common/target_plugin_loader.c
@@ -56,6 +56,7 @@ static struct xrt_plugin_instance *g_active_instance = NULL;
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
 #include <stdio.h>
+#include <wchar.h>
 
 /*!
  * Maximum number of plug-ins we enumerate from the registry. 16 is
@@ -295,10 +296,95 @@ try_load_one(const struct plugin_entry *e, struct xrt_plugin_instance **out_inst
 	return iface;
 }
 
+/*!
+ * Pin the runtime core DLL (DisplayXRClient.dll) into the process by
+ * absolute path before any plug-in is loaded. Runs once.
+ *
+ * Vendor plug-in DLLs statically import DisplayXRClient.dll, which ships
+ * in the runtime install dir. That dir is intentionally NOT on PATH
+ * (issue #104), and the per-plug-in LOAD_WITH_ALTERED_SEARCH_PATH below
+ * searches only the plug-in's own dir + System32 + cwd + PATH — none of
+ * which hold DisplayXRClient.dll except cwd when it happens to equal the
+ * runtime dir. So at logon (Run-key launches with cwd=System32) and when
+ * the workspace controller spawns the service, that import fails with
+ * ERROR_MOD_NOT_FOUND (126) and every plug-in load aborts → no display
+ * processor → the service exits (issue #328).
+ *
+ * Pre-loading DisplayXRClient.dll by absolute path here pins it into the
+ * process by base name; the loader then satisfies each plug-in's import
+ * from the already-resident module with no disk search, independent of
+ * cwd/PATH. The per-plug-in load deliberately keeps
+ * LOAD_WITH_ALTERED_SEARCH_PATH so vendor side DLLs (e.g. the SR platform
+ * at C:\Program Files\LeiaSR\Platform\bin) still resolve via PATH.
+ *
+ * No process-wide search-order mutation (cf. target_dll_init.c) — this is
+ * the surgical option and leaves PATH resolution intact for vendors.
+ */
+static void
+preload_runtime_core_dll(void)
+{
+	static int done = 0;
+	if (done) {
+		return;
+	}
+	done = 1;
+
+	// Directory of whatever module this loader code lives in — the
+	// service exe or DisplayXRClient.dll, both of which sit in the
+	// runtime install dir. UNCHANGED_REFCOUNT: we only want its path.
+	HMODULE self = NULL;
+	if (!GetModuleHandleExW(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS |
+	                            GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
+	                        (LPCWSTR)(void *)&preload_runtime_core_dll, &self)) {
+		U_LOG_W("plugin loader: GetModuleHandleEx(self) failed (err=%lu) — cannot locate runtime dir to "
+		        "pre-load DisplayXRClient.dll.",
+		        GetLastError());
+		return;
+	}
+
+	wchar_t modpath[MAX_PATH];
+	DWORD len = GetModuleFileNameW(self, modpath, MAX_PATH);
+	if (len == 0 || len >= MAX_PATH) {
+		U_LOG_W("plugin loader: GetModuleFileName(self) failed/truncated — skipping DisplayXRClient.dll "
+		        "pre-load.");
+		return;
+	}
+
+	// Strip the module filename to leave the runtime dir.
+	wchar_t *slash = wcsrchr(modpath, L'\\');
+	if (slash == NULL) {
+		return;
+	}
+	*slash = L'\0';
+
+	wchar_t corepath[MAX_PATH];
+	if (swprintf_s(corepath, MAX_PATH, L"%s\\DisplayXRClient.dll", modpath) <= 0) {
+		return;
+	}
+
+	// If it is already resident (in-process app target, where the host
+	// loaded it via the OpenXR loader) this just bumps the refcount and
+	// the plug-in import binds to it. Otherwise it loads now so the
+	// import resolves. LOAD_WITH_ALTERED_SEARCH_PATH so the core DLL's
+	// own siblings (cjson.dll, pthreadVC3.dll) resolve from the runtime
+	// dir rather than the host exe dir.
+	if (LoadLibraryExW(corepath, NULL, LOAD_WITH_ALTERED_SEARCH_PATH) == NULL) {
+		U_LOG_W("plugin loader: pre-load of %ls failed (err=%lu); plug-in loads may fail when cwd is not "
+		        "the runtime dir.",
+		        corepath, GetLastError());
+	} else {
+		U_LOG_I("plugin loader: pinned runtime core DLL for plug-in import resolution: %ls", corepath);
+	}
+}
+
 static const struct xrt_plugin_iface *
 discover_active_plugin(struct xrt_plugin_instance **out_inst)
 {
 	*out_inst = NULL;
+
+	// Ensure DisplayXRClient.dll is resident before any plug-in import
+	// of it has to resolve (issue #328).
+	preload_runtime_core_dll();
 
 	struct plugin_entry entries[MAX_PLUGIN_ENTRIES];
 	int n = enumerate_registry(entries, MAX_PLUGIN_ENTRIES);

